Bottom: 6d36e6ada8c0350a21d793fb809595f6ad64ebd5
Top:    9bc796d862a35d27a0a8119e6e1567e335f9c31e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-27 17:55:04 +0100

Refresh of shiny-3d-eye-candy-0

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index e473df8..f5e508c 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1628,14 +1628,14 @@ ghid_event_to_pcb_coords (int event_x, int event_y, Coord *pcb_x, Coord *pcb_y)
 bool
 ghid_pcb_to_event_coords (Coord pcb_x, Coord pcb_y, int *event_x, int *event_y)
 {
-  *event_x = view_matrix[0][0] * pcb_x +
-             view_matrix[0][1] * pcb_y +
-             view_matrix[0][2] * 0 +
-             view_matrix[0][3] * 1;
-  *event_y = view_matrix[1][0] * pcb_x +
-             view_matrix[1][1] * pcb_y +
-             view_matrix[1][2] * 0 +
-             view_matrix[1][3] * 1;
+  *event_x = view_matrix[0][0] * (float)pcb_x +
+             view_matrix[0][1] * (float)pcb_y +
+             view_matrix[0][2] * global_depth +
+             view_matrix[0][3] * 1.;
+  *event_y = view_matrix[1][0] * (float)pcb_x +
+             view_matrix[1][1] * (float)pcb_y +
+             view_matrix[1][2] * global_depth +
+             view_matrix[1][3] * 1.;
 
   return true;
 }
